File: Devide&Conquer/1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        if n < 0:
            return 1 / self.myPow(x, -n)
        if n % 2:
            return x * self.myPow(x, n - 1)
        return self.myPow(x * x, n // 2)

class Solution:
    def myPow(self, x: float, n: int) -> float:
        def Mul(N):
            ans = 1.0
            _x = x
            while N:
                if N & 1:
                    ans *= _x
                _x *= _x
                N >>= 1
            return ans
        return Mul(n) if n >= 0 else 1 / Mul(-n)

File: weekGames/DWeekGame_134/1.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors) {
        int ans = 0, n = colors.size();
        for (int i = 1; i < n - 1; ++i) {
            if(colors[i] != colors[i - 1] && colors[i] != colors[i + 1]) ++ans;
        }
        if(colors[0] != colors[1] && colors[0] != colors[n - 1]) ++ans;
        if(colors[n - 1] != colors[0] && colors[n - 1] != colors[n - 2]) ++ans;
        return ans;
    }
};

static auto _ = []() {
    cin.tie(0);
    ios_base::sync_with_stdio(false);
    return 0;
}();